File: backend/src/main/java/com/quyenauto/lead/dto/LeadResponse.java
package com.quyenauto.lead.dto;

import lombok.Builder;
import lombok.Data;

import java.time.LocalDateTime;

@Data
@Builder
public class LeadResponse {
    private Long id;
    private String phone;
    private String name;
    private Long productId;
    private String productName;
    private String note;
    private String specifications;
    private boolean contacted;
    private LocalDateTime createdAt;
}

File: lib/data/models/response/lead_response.dart
class LeadResponse {
  final int id;
  final String phone;
  final String? name;
  final int? productId;
  final String? productName;
  final String? note;
  final String? specifications;
  final bool contacted;
  final DateTime createdAt;

  const LeadResponse({
    required this.id,
    required this.phone,
    this.name,
    this.productId,
    this.productName,
    this.note,
    this.specifications,
    required this.contacted,
    required this.createdAt,
  });

  factory LeadResponse.fromJson(Map<String, dynamic> json) => LeadResponse(
        id: (json['id'] as num).toInt(),
        phone: json['phone'] as String,
        name: json['name'] as String?,
        productId: (json['productId'] as num?)?.toInt(),
        productName: json['productName'] as String?,
        note: json['note'] as String?,
        specifications: json['specifications'] as String?,
        contacted: json['contacted'] as bool? ?? false,
        createdAt: DateTime.parse(json['createdAt'] as String),
      );
}
